cfme/infrastructure/config_management.py

```diff
@@ -259,37 +181,37 @@ def delete(self, cancel=False, wait_deleted=True, force=False):
         """
         if not force and not self.exists:
             return
-        sel.force_navigate('infrastructure_config_manager_remove', context={'manager': self})
+        navigate_to(self, 'All')
+        sel.check(Quadicon(self.quad_name, None).checkbox())
+        cfg_btn('Remove selected items from the VMDB', invokes_alert=True)
+
         sel.handle_alert(cancel)
         if not cancel:
-            flash_msg = version.pick({
-                version.LOWEST: 'Delete initiated for 1 Provider from the CFME Database',
-                '5.6': 'Delete initiated for 1 provider'
-            })
+            flash_msg = 'Delete initiated for 1 provider'
             flash.assert_message_match(flash_msg)
             if wait_deleted:
                 wait_for(func=lambda: self.exists, fail_condition=True, delay=15, num_sec=60)
 
     @property
     def _refresh_flash_msg(self):
-        return version.pick({
-            version.LOWEST: 'Refresh {0} initiated for 1 Provider ({0}) from the CFME Database'
-            .format(self.type),
-            '5.6': 'Refresh Provider initiated for 1 provider ({})'.format(self.type)
-        })
+        return 'Refresh Provider initiated for 1 provider ({})'.format(self.type)
 
     @property
     def exists(self):
         """Returns whether the manager exists in the UI or not"""
-        sel.force_navigate('infrastructure_config_managers')
+        navigate_to(self, 'All')
         if (Quadicon.any_present() and
```

Comment:
you're right. fixed. 
But to be honest I think this's bad idea to do two things at the same time - refactoring of old code along with movement to navmazing.
It not only takes time but may cause side issues.